.github/workflows/fetch-release-data.yml

```diff
@@ -6,25 +6,67 @@ on:
   workflow_dispatch:
 permissions:
   contents: write
+  pull-requests: write
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  BRANCH: fetch-release-data-update
+
 jobs:
   fetch-release-data:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
+      - name: Try checkout ${{ env.BRANCH }}
+        uses: actions/checkout@v4
+        id: try-checkout
+        continue-on-error: true
         with:
-          sparse-checkout: |
-            scripts
-            static/data/releases
-      - run: |
+          ref: ${{ env.BRANCH }}
+
+      - name: Fallback to default branch if checkout failed
+        if: steps.try-checkout.outcome == 'failure'
```

Comment:
The potential problem with this approach is that once you create `fetch-release-data-update` branch and forget to delete it, this workflow will always use that instead of default branch. Might be fine but please document this behavior somewhere.